File: languages/python/bloock/entity/identity/integer_attribute.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto
from bloock.entity.identity.attribute import Attribute


class IntegerAttribute(Attribute):
    """
    Represents an attribute with an integer value.
    """
    @staticmethod
    def from_proto(a: proto.IntegerAttribute) -> IntegerAttribute:
        return IntegerAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.IntegerAttribute:
        return proto.IntegerAttribute(
            id=self.id,
            value=self.value,
        )

File: languages/php/Bloock/Entity/Identity/IntegerAttribute.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents an attribute with an integer value.
 */
class IntegerAttribute extends Attribute
{
    public static function fromProto(\Bloock\IntegerAttribute $res): IntegerAttribute
    {
        return new IntegerAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\IntegerAttribute
    {
        $p = new \Bloock\IntegerAttribute();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}